oml/miners/inbatch_top_pn.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from oml.interfaces.miners import ITripletsMinerInBatch, TTripletsIds
+from oml.utils.misc_torch import pairwise_dist
+
+
+class TopPNTripletsMiner(ITripletsMinerInBatch):
+    """
+    This miner selects top hard triplets based on distances between features:
+
+    - hard `positive` samples has large distance to the anchor sample
+
+    - hard `negative` samples has small distance to the anchor sample
+
+    Toward the end of the training, annotation errors can affect final metric. If you are not sure about
+    the quality of your dataset, you can use range instead of integer value for paramters and exclude combinations
+    with the largest distances. For example instead picking `top5` positive examples, you can use `top2 - top5` examples
+
+    """
+
+    def __init__(
+        self,
+        top_positive: Union[Tuple[int, int], List[int], int] = 1,
+        top_negative: Union[Tuple[int, int], List[int], int] = 1,
```

Comment:
we need assert that tops have to be more or equal than 1

Reply:
n_positive, n_negative

Reply:
in parser